x-pack/plugins/ml/public/explorer/explorer_charts/explorer_chart_distribution.js

```diff
@@ -0,0 +1,463 @@
+        .call(yAxis);
+
+      if (tooManyBuckets === false) {
+        removeLabelOverlap(gAxis, emphasisStart, interval, vizWidth);
+      }
+    }
+
+    function drawRareChartDots(dotsData, rareChartGroup, rareChartValuesLine, radius = 1.5) {
+      // check if `g.values-dots` already exists, if not create it
+      // in both cases assign the element to `dotGroup`
+      const dotGroup = (rareChartGroup.select('.values-dots').empty())
+        ? rareChartGroup.append('g').classed('values-dots', true)
+        : rareChartGroup.select('.values-dots');
+
+      // use d3's enter/update/exit pattern to render the dots
+      const dots = dotGroup.selectAll('circle').data(dotsData);
+
+      dots.enter().append('circle')
+        .classed('values-dots-circle', true)
+        .classed('values-dots-circle-blur', (d) => {
+          return (d.entity !== highlight);
+        })
+        .attr('r', d => ((d.entity === highlight) ? (radius * 1.5) : radius));
+
+      dots
+        .attr('cx', rareChartValuesLine.x())
+        .attr('cy', rareChartValuesLine.y());
+
+      dots.exit().remove();
+    }
+
+    function drawRareChartHighlightedSpan() {
+      // Draws a rectangle which highlights the time span that has been selected for view.
+      // Note depending on the overall time range and the bucket span, the selected time
+      // span may be longer than the range actually being plotted.
+      const rectStart = Math.max(config.selectedEarliest, config.plotEarliest);
+      const rectEnd = Math.min(config.selectedLatest, config.plotLatest);
+      const rectWidth = lineChartXScale(rectEnd) - lineChartXScale(rectStart);
+
+      lineChartGroup.append('rect')
+        .attr('class', 'selected-interval')
+        .attr('x', lineChartXScale(new Date(rectStart)) + 2)
+        .attr('y', 2)
+        .attr('rx', 3)
+        .attr('ry', 3)
+        .attr('width', rectWidth - 4)
+        .attr('height', chartHeight - 4);
+    }
+
+    function drawRareChartMarkers(data) {
+      // Render circle markers for the points.
+      // These are used for displaying tooltips on mouseover.
+      // Don't render dots where value=null (data gaps)
+      const dots = lineChartGroup.append('g')
+        .attr('class', 'chart-markers')
+        .selectAll('.metric-value')
+        .data(data.filter(d => d.value !== null));
+
+      // Remove dots that are no longer needed i.e. if number of chart points has decreased.
+      dots.exit().remove();
+      // Create any new dots that are needed i.e. if number of chart points has increased.
+      dots.enter().append('circle')
+        .attr('r', LINE_CHART_ANOMALY_RADIUS)
+        .on('mouseover', function (d) {
+          showLineChartTooltip(d, this);
+        })
+        .on('mouseout', () => mlChartTooltipService.hide());
+
+      // Update all dots to new positions.
+      const threshold = mlSelectSeverityService.state.get('threshold');
+      dots.attr('cx', function (d) { return lineChartXScale(d.date); })
```

Comment:
nit, looks like these could be arrow functions?
`dots.attr('cx', d => lineChartXScale(d.date))`

Reply:
thanks, will do that in the follow up when consolidating charts code, took a note in #21163